CMakeLists.txt

```diff
@@ -119,7 +119,8 @@ if(LIBUTP_BUILD_PROGRAMS AND NOT MSVC)
     target_compile_definitions(ucat
         PRIVATE
             $<$<NOT:$<BOOL:${WIN32}>>:_POSIX_C_SOURCE=200112L>
-            $<$<BOOL:${APPLE}>:_DARWIN_C_SOURCE>)
+            $<$<BOOL:${APPLE}>:_DARWIN_C_SOURCE>
+            $<$<BOOL:${BSD}>:__BSD_VISIBLE>)
```

Comment:
Any names starting with two underscores are internal to the implementation the way I see it. Is there another macro that works, e.g. `_BSD_SOURCE` (seems to be deprecated) or `_DEFAULT_SOURCE`?

Reply:
Yes, it's bad form. If I understand correctly, you're forcing POSIX types on netinet/ip.h, which is not a POSIX header. Unfortunately, FreeBSD recognizes neither _BSD_SOURCE nor _DEFAULT_SOURCE.

My understanding is that setting POSIX is not necessary on the BSDs to get the various benefits it normally carries. [1](https://github.com/kcat/openal-soft/issues/4#issuecomment-100038076). For that reason, one alternative is to exclude the BSDs from POSIX along with Windows. It builds fine under that scenario.

Reply:
Yeah, not defining `_POSIX_C_SOURCE` sounds like a better idea. If you want to amend the PR, I'd appreciate it.